library/swift/test/RequestBuilderTests.swift

```diff
@@ -0,0 +1,189 @@
+    XCTAssertNil(request.headers["foo"])
+  }
+
+  func testRemovingSpecificHeaderKeyDoesNotRemoveOtherKeysFromRequest() {
+    let request = RequestBuilder(method: .post, url: kURL)
+      .addHeader(name: "foo", value: "1")
+      .addHeader(name: "bar", value: "2")
+      .removeHeaders(name: "foo")
+      .build()
+    XCTAssertEqual(["bar": ["2"]], request.headers)
+  }
+
+  func testRemovingSpecificHeaderValueRemovesItFromRequest() {
+    let request = RequestBuilder(method: .post, url: kURL)
+      .addHeader(name: "foo", value: "1")
+      .addHeader(name: "foo", value: "2")
+      .addHeader(name: "foo", value: "3")
+      .removeHeader(name: "foo", value: "2")
+      .build()
+    XCTAssertEqual(["1", "3"], request.headers["foo"])
+  }
+
+  func testRemovingAllHeaderValuesRemovesKeyFromRequest() {
+    let request = RequestBuilder(method: .post, url: kURL)
+      .addHeader(name: "foo", value: "1")
+      .addHeader(name: "foo", value: "2")
+      .removeHeader(name: "foo", value: "1")
+      .removeHeader(name: "foo", value: "2")
+      .build()
+    XCTAssertNil(request.headers["foo"])
+  }
+
+  // MARK: - Trailers
+
+  func testAddingNewTrailerAppendsToListOfTrailerKeys() {
+    let request = RequestBuilder(method: .post, url: kURL)
+      .addTrailer(name: "foo", value: "bar")
+      .build()
+    XCTAssertEqual(["bar"], request.trailers["foo"])
+  }
+
+  func testRemovingSpecificTrailerKeyRemovesAllOfItsValuesFromRequest() {
+    let request = RequestBuilder(method: .post, url: kURL)
+      .addTrailer(name: "foo", value: "1")
+      .addTrailer(name: "foo", value: "2")
+      .removeTrailers(name: "foo")
+      .build()
+    XCTAssertNil(request.trailers["foo"])
+  }
+
+  func testRemovingSpecificTrailerKeyDoesNotRemoveOtherKeysFromRequest() {
+    let request = RequestBuilder(method: .post, url: kURL)
+      .addTrailer(name: "foo", value: "1")
+      .addTrailer(name: "bar", value: "2")
+      .removeTrailers(name: "foo")
+      .build()
+    XCTAssertEqual(["bar": ["2"]], request.trailers)
+  }
+
+  func testRemovingSpecificTrailerValueRemovesItFromRequest() {
+    let request = RequestBuilder(method: .post, url: kURL)
+      .addTrailer(name: "foo", value: "1")
+      .addTrailer(name: "foo", value: "2")
+      .addTrailer(name: "foo", value: "3")
+      .removeTrailer(name: "foo", value: "2")
+      .build()
+    XCTAssertEqual(["1", "3"], request.trailers["foo"])
+  }
+
+  func testRemovingAllTrailerValuesRemovesKeyFromRequest() {
+    let request = RequestBuilder(method: .post, url: kURL)
+      .addTrailer(name: "foo", value: "1")
+      .addTrailer(name: "foo", value: "2")
+      .removeTrailer(name: "foo", value: "1")
+      .removeTrailer(name: "foo", value: "2")
+      .build()
+    XCTAssertNil(request.trailers["foo"])
+  }
+
+  // MARK: - Request conversion
+
+  func testRequestsAreEqualWhenPropertiesAreEqual() {
+    let request1 = kRequestBuilder1.build()
+    let request2 = kRequestBuilder2.build()
```

Comment:
Shouldn't you use `kRequestBuilder1` for both? Cause if we end up modifying `kRequestBuilder2` we'll break this test no?

Reply:
I originally duplicated this since re-using `kRequestBuilder1` would mean that the builder would have the same memory address. This would result in this test passing even if the properties were _not_ the same. Updated to add a helper function for creating new (identical) builders with different memory addresses 👍